daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_functions/BitFunctions.tdml

```diff
@@ -0,0 +1,726 @@
+                            <value>4</value>
+                        </ex:rightByte1>
+                        <ex:rightByte2>
+                            <num>4</num>
+                            <count>7</count>
+                            <value>0</value>
+                        </ex:rightByte2>
+                  </ex:testByteShifting>
+              </tdml:dfdlInfoset>
+          </tdml:infoset>
+    </tdml:parserTestCase>
+
+    <tdml:parserTestCase name="shortShiftTest" root="testShortShifting" model="BitFunctions">
+      <tdml:document>4,2|4,7|4,0|4,7</tdml:document>
+          <tdml:infoset>
+              <tdml:dfdlInfoset>
+                  <ex:testShortShifting>
+                        <ex:leftShort1>
+                            <num>4</num>
+                            <count>2</count>
+                            <value>16</value>
+                        </ex:leftShort1>
+                        <ex:leftShort2>
+                            <num>4</num>
+                            <count>7</count>
+                            <value>512</value>
+                        </ex:leftShort2>
+                        <ex:rightShort1>
+                            <num>4</num>
+                            <count>0</count>
+                            <value>4</value>
+                        </ex:rightShort1>
+                        <ex:rightShort2>
+                            <num>4</num>
+                            <count>7</count>
+                            <value>0</value>
+                        </ex:rightShort2>
+                  </ex:testShortShifting>
+              </tdml:dfdlInfoset>
+          </tdml:infoset>
+    </tdml:parserTestCase>
+
+    <tdml:parserTestCase name="testUnsignedLongShifting1" root="testUnsignedLongShifting" model="BitFunctions">
+      <tdml:document>1,50|18446744073709551,0|4,0|1,50</tdml:document>
+          <tdml:infoset>
+              <tdml:dfdlInfoset>
+                  <ex:testUnsignedLongShifting>
+                        <ex:leftUnsignedLong1>
+                            <num>1</num>
+                            <count>50</count>
+                            <value>1125899906842624</value>
+                        </ex:leftUnsignedLong1>
+                        <ex:leftUnsignedLong2>
+                            <num>18446744073709551</num>
+                            <count>0</count>
+                            <value>18446744073709551</value>
+                        </ex:leftUnsignedLong2>
+                        <ex:rightUnsignedLong1>
+                            <num>4</num>
+                            <count>0</count>
+                            <value>4</value>
+                        </ex:rightUnsignedLong1>
+                        <ex:rightUnsignedLong2>
+                            <num>1</num>
+                            <count>50</count>
+                            <value>1125899906842624</value>
+                        </ex:rightUnsignedLong2>
```

Comment:
Please add more tests for unsignedLong. This is the one that is different from all the others in the implementation.

Please cover 
* shifting left the ULong.MaxValue (18446744073709551615) by 63  is 9223372036854775808 (0x8000000000000000)
* shifting right the ULong.MaxValue by 63 is 1

Reply:
If you rebase your branch, the comment history here will be lost. 

The right sequence is: 
fix everything here in hopefully just one more commit. 

When you get two +1 approvals, then ...

1.  create a safety branch e.g., "git branch DAFFODIL-2473-save". That will save your current branch with all the separate commits as this new branch. 
2. Then you can squash all your commits together into one commit. (I usually do this at command line with git rebase -i . There may be some graphical tool for this)
3. fetch the latest master branch commits of daffodil, and insure your local master branch is at that point.
4. rebase your one new squashed commit onto the latest master, (which now has the ULong fixes I did btw.)
4. test throughly with sbt test, and sbt it:test, which should produce no errors,
5. then force push to this PR branch again.